src/Nethermind/Nethermind.TxPool.Test/Collections/DistinctValueSortedPoolTests.cs

```diff
@@ -155,23 +168,23 @@ public WithFinalizerDistinctPool(int capacity, IComparer<WithFinalizer> comparer
             protected override int GetKey(WithFinalizer value) => value.Index;
         }
 
-        [Test]
-        public void Capacity_is_never_exceeded()
+        IComparer<WithFinalizer> _comparer = Comparer<WithFinalizer>.Create((t1, t2) =>
         {
-            IComparer<WithFinalizer> comparer = Comparer<WithFinalizer>.Create((t1, t2) =>
-            {
-                int t1Oddity = t1.Index % 2;
-                int t2Oddity = t2.Index % 2;
+            int t1Oddity = t1.Index % 2;
+            int t2Oddity = t2.Index % 2;
 
-                if (t1Oddity.CompareTo(t2Oddity) != 0)
-                {
-                    return t1Oddity.CompareTo(t2Oddity);
-                }
+            if (t1Oddity.CompareTo(t2Oddity) != 0)
+            {
+                return t1Oddity.CompareTo(t2Oddity);
+            }
 
-                return t1.Index.CompareTo(t2.Index);
-            });
+            return t1.Index.CompareTo(t2.Index);
+        });
 
-            var pool = new WithFinalizerDistinctPool(Capacity, comparer, new WithFinalizerComparer(), LimboLogs.Instance);
+        [Test]
+        public void Capacity_is_never_exceeded()
```

Comment:
Should this run in parallel or should their be a Parallel varient of it? (Since txPool is running concurrently with adds)

Reply:
there is Capacity_is_never_exceeded_with_multiple_threads
and I've just refactored Capacity_is_never_exceeded